scripts/eqtl_hail_batch/generate_eqtl_spearman.py

```diff
@@ -0,0 +1,449 @@
+#!/usr/bin/env python3
+
+"""Run Spearman rank correlation on SNPs and expression residuals"""
+
+import os
+
+import hail as hl
+import hailtop.batch as hb
+import pandas as pd
+import numpy as np
+import statsmodels.api as sm
+import statsmodels.stats.multitest as multi
+from patsy import dmatrices  # pylint: disable=no-name-in-module
+from scipy.stats import spearmanr
+from cpg_utils.hail_batch import (
+    dataset_path,
+    output_path,
+    copy_common_env,
+    init_batch,
+    remote_tmpdir,
+)
+from cloudpathlib import AnyPath
+import click
+
+DEFAULT_DRIVER_MEMORY = '4G'
+DRIVER_IMAGE = os.getenv('CPG_DRIVER_IMAGE')
+assert DRIVER_IMAGE
+
+TOB_WGS = dataset_path('mt/v7.mt/')
+FREQ_TABLE = dataset_path('joint-calling/v7/variant_qc/frequencies.ht/', 'analysis')
+
+
+def filter_lowly_expressed_genes(expression_df):
+    """Remove genes with low expression in all samples"""
+
+    # Remove genes with 0 expression in all samples
+    expression_df = expression_df.loc[:, (expression_df != 0).any(axis=0)]
+    genes_not_equal_zero = expression_df.iloc[:, 1:].values != 0
+    n_expr_over_zero = pd.DataFrame(genes_not_equal_zero.sum(axis=0))
+    percent_expr_over_zero = (n_expr_over_zero / len(expression_df.index)) * 100
+    percent_expr_over_zero.index = expression_df.columns[1:]
+
+    # Filter genes with less than 10 percent individuals with non-zero expression
+    atleast10percent = percent_expr_over_zero[(percent_expr_over_zero > 10)[0]]
+    sample_ids = expression_df['sampleid']
+    expression_df = expression_df[atleast10percent.index]
+    expression_df.insert(loc=0, column='sampleid', value=sample_ids)
+
+    return expression_df
+
+
+def get_number_of_scatters(expression_df, geneloc_df):
+    """get index of total number of genes"""
+
+    expression_df = filter_lowly_expressed_genes(expression_df)
+    gene_ids = list(expression_df.columns.values)[1:]
+    geneloc_df = geneloc_df[geneloc_df.gene_name.isin(gene_ids)]
+    # subset genes to 10 test genes
+    # remove this when running on entire dataset
+    genes_to_test = ['ERAP2', 'LAMB1', 'POLR2J2', 'GBP7', 'GSTM3', 'CCDC163P', 'CTLA4', 'AIF1', 'BRCA1', 'IL7', 'HDHD5']
```

Comment:
Should this still be here?

Reply:
This actually does stay - I'm currently only testing 10 genes, but this will change later on